libtenzir/builtins/aggregation-functions/sample.cpp

```diff
@@ -67,6 +68,13 @@ class plugin : public virtual aggregation_function_plugin {
   auto aggregation_default() const -> data override {
     return caf::none;
   }
+
+  auto make_aggregation(invocation inv, session ctx) const
+    -> failure_or<std::unique_ptr<aggregation_instance>> override {
+    const auto* first = plugins::find<aggregation_plugin>("first");
+    TENZIR_ASSERT(first);
+    return first->make_aggregation(invocation{inv.call}, ctx);
```

Comment:
Actually, would you mind adding a deprecation warning here?

Reply:
```diff
diff --git a/libtenzir/builtins/aggregation-functions/sample.cpp b/libtenzir/builtins/aggregation-functions/sample.cpp
index a8341b5b69f..0f4c23e2968 100644
--- a/libtenzir/builtins/aggregation-functions/sample.cpp
+++ b/libtenzir/builtins/aggregation-functions/sample.cpp
@@ -71,6 +71,9 @@ class plugin : public virtual aggregation_function_plugin,
 
   auto make_aggregation(invocation inv, session ctx) const
     -> failure_or<std::unique_ptr<aggregation_instance>> override {
+    diagnostic::warning("deprecated: use `first` instead")
+      .primary(inv.call)
+      .emit(ctx);
     const auto* first = plugins::find<aggregation_plugin>("first");
     TENZIR_ASSERT(first);
     return first->make_aggregation(invocation{inv.call}, ctx);
```

This revealed an interesting bug: parse-time diagnostics from aggregation functions always get emitted twice. I don't think that should be the case.

Reply:
Oh, yes. That's a regression from https://github.com/tenzir/tenzir/pull/4527, caused by moving the deduplicator into the inner execution function.